modules/eks/cluster/CHANGELOG.md

```diff
@@ -1,4 +1,166 @@
+Components PR [#852](https://github.com/cloudposse/terraform-aws-components/pull/852)
+
+This is a bug fix and feature enhancement update. No action is necessary to upgrade.
+However, with the new features and new recommendations, you may want to change
+your configuration.
+
+## Recommended (optional) changes
+
+Previously, we recommended deploying Karpenter to Fargate and not provisioning
+any nodes. However, this causes issues with add-ons that require compute power
+to fully initialize, such as `coredns`, and it can reduce the cluster to a
+single node, removing the high availability that comes from having a node
+per Availability Zone and replicas of pods spread across those nodes.
+
+As a result, we now recommend deploying a minimal node group with a single
+instance (currently recommended to be a `c6a.large`) in each of 3 Availability
+Zones. This will provide the compute power needed to initialize add-ons, and
+will provide high availability for the cluster. As a bonus, it will also
+remove the need to deploy Karpenter to Fargate.
+
+**NOTE about instance type**: The `c6a.large` instance type is relatively
+new. If you have deployed an old version of our ServiceControlPolicy
+`DenyEC2NonNitroInstances`, `DenyNonNitroInstances` (obsolete, replaced by
+`DenyEC2NonNitroInstances`), and/or`DenyEC2InstancesWithoutEncryptionInTransit`,
+you will want to update them to v0.12.0 or choose a difference instance type.
+
+### Migration procedure
+
+To perform the recommended migration, follow these steps:
+
+#### 1. Deploy a minimal node group, move addons to it
+
+Change your `eks/cluster` configuration to set `deploy_addons_to_fargate: false`.
+
+Add the following to your `eks/cluster` configuration, but
+copy the block device name, volume size, and volume type from your existing
+Karpenter provisioner configuration. Also select the correct `ami_type`
+according to the `ami_family` in your Karpenter provisioner configuration.
+
+```yaml
+  node_groups:
+    # will create 1 node group for each item in map
+    # Provision a minimal static node group for add-ons and redundant replicas
+    main:
+      # EKS AMI version to use, e.g. "1.16.13-20200821" (no "v").
+      ami_release_version: null
+      # Type of Amazon Machine Image (AMI) associated with the EKS Node Group
+      ami_type: BOTTLEROCKET_x86_64
+      # Additional name attributes (e.g. `1`) for the node group
+      attributes: []
+      # will create 1 auto scaling group in each specified availability zone
+      # or all AZs with subnets if none are specified anywhere
+      availability_zones: null
+      # Whether to enable Node Group to scale its AutoScaling Group
+      cluster_autoscaler_enabled: false
+      # True (recommended) to create new node_groups before deleting old ones, avoiding a temporary outage
+      create_before_destroy: true
+      # Configure storage for the root block device for instances in the Auto Scaling Group
+      block_device_map:
+        "/dev/xvdb":
```

Comment:
The default `/dev/xvda` is sufficient for Bottlerocket and does not need to be configured. Since only the kernel uses it, it does not run out of space with Kubernetes usage.